File: examples/ruby/bad.rb
# @docdeps: allowed_imports = ["set"]
require "json"

# @docslim: max_lines = 3, max_nested_depth = 1, max_complexity = 2
# @docpure: deterministic = true
def handle(amount, d_items)
  puts "side effect"
  if amount
    if amount > 1
      return d_items.length
    end
  end
  0
end

File: examples/php/bad.php
<?php
// @docdeps: allowed_imports = ["App\\Crypto"]
use App\Logger;

// @docslim: max_lines = 3, max_nested_depth = 1, max_complexity = 2
// @docpure: deterministic = true
function handle(int $amount, array $d_items): int {
  echo "side effect";
  if ($amount) {
    if ($amount > 1) {
      return count($d_items);
    }
  }
  return 0;
}
